Continue work on Unified release notes

h-vetinari opened the pull request:
Follow-up to #5526

h-vetinari commented on CHANGES.rst:
```diff
@@ -243,12 +278,170 @@ Cython 3.0.0 also aligns many semantics with Python 3, in particular:
 
 Pure python mode gained many new features and was generally overhauled to make
-it as capable as the 
+it as capable as the
```
This sentence (added in https://github.com/cython/cython/pull/5526/commits/66a542073dd84f29a1eab065938f90c3e23e6fa3) is still dangling

h-vetinari commented on CHANGES.rst:
```diff
@@ -234,6 +263,12 @@ Related fixes
+is available through the directive ``language_level=2``.
+(Github issue :issue:`2565`)
+
 Cython 3.0.0 also aligns many semantics with Python 3, in particular:
 
 * division
```
Paging @da-woods for a summary :)

scoder commented:
Sorry for the conflict. Since a large part of the changelog was duplicated (I had thought that I had already removed that part but apparently hadn't), I removed in in the master branch in order to avoid confusion and kept it only here. I hope it worked out this time.

h-vetinari commented:
> Since a large part of the changelog was duplicated (I had thought that I had already removed that part but apparently hadn't), I removed in in the master branch in order to avoid confusion and kept it only here

Note that https://github.com/cython/cython/commit/2428384dc3283b6d7edb2950a535116672c0b599 does not remove all the duplication, just those entries that hadn't been moved to the unified release notes yet -- it was intentional to track those so that I don't end up through the various C&P moves so I don't end up accidentally deleting something. That's why I wanted to continue work on #5526 a bit longer 😅 

Now the global diff is quite unreadable, but I'll rebase on a revert of that commit to fix it, so that it becomes clearer what's happening.

h-vetinari commented on CHANGES.rst:
```diff
@@ -389,199 +584,1034 @@ Related fixes
+  to make it more visible.
+
+
+vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
+vvv    TO DISTRIBUTE TO PROPER SECTIONS ABOVE    vvv
+vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
```
Everything below this still needs to be moved.

h-vetinari commented:
OK, so I removed the "duplicate" part in https://github.com/cython/cython/pull/5532/commits/7245b4a0106d59dddaafa194a6b42f4eee83ffba for now (can revert for the next PR), because IMO a lot of the changes in this PR should have already gone in with #5526.

Could you PTAL and maybe merge @scoder? I can then open a PR that tries dealing with the rest.

h-vetinari commented:
Well, the release notes lose importance with every passing day, so I guess it's time to give up this effort (it's a pity that the release notes aspect felt so rushed vis-à-vis the enormous amount of stuff in 3.0 deserving more of a spotlight). For the record, it's not a nice experience to merge a PR that the contributor says is not ready (causing me to scramble to fix it), and then turn to something else.
